compiler/rustc_parse/src/parser/mod.rs

```diff
@@ -1126,6 +1127,16 @@ impl<'a> Parser<'a> {
         }
     }
 
+    /// Parses genness: `gen` or nothing.
+    fn parse_genness(&mut self, case: Case) -> Gen {
+        if self.token.span.at_least_rust_2024() && self.eat_keyword_case(kw::Gen, case) {
```

Comment:
`parse_asyncness` doesn't check the span's edition -- why not?

Reply:
Just an oversight. I have since started checking the edition at the `parse_genness` call sites

Reply:
Ah... this avoids emitting

> expected one of `async`, `extern`, `fn`, `gen`, or `unsafe`, found keyword `pub`'

on older editions